test/randomized_benchmarking/test_rb_utils.py

```diff
@@ -204,14 +210,12 @@ def test_clifford_1_qubit_generation(self):
             {"stabilizer": ["-Y"], "destabilizer": ["+Z"]},
         ]
         cliffords = [Clifford.from_dict(i) for i in clifford_dicts]
-        utils = rb.CliffordUtils()
         for n in range(24):
-            clifford = utils.clifford_1_qubit(n)
+            clifford = CliffordUtils.clifford_1_qubit(n)
```

Comment:
I think it makes sense to split `test_rb_utils` into 2 files: `test_rb_utils` and `test_clifford_utils`, which will contain only the tests for cliffords.

In addition, since this PR mainly introduces a new technique to compose and inverse cliffords, I would suggest to add the tests `test_compose_cliffords` and `tests_inverse_clifford` that will generate pseudo-random cliffords (using a fixed seed) and compose/inverse them using the clifford class methods and the new method, and verify that one gets the same output.

Reply:
ok and ok. Will do both.

Reply:
done.